src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,12 +1048,25 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static ReadOnlyCollection<SyntaxKind> _lazyContextualKeywordKindsArray;
```

Comment:
If it isn't needed, then why are we changing it? :)

Let's step back a second.  In general, PRs need to demontrate their value and a reason for them to be made.  I understood the value in you change to change the constant so that you could add contextual keywords withou thaving to update this.  However, no data or reasoning was given as to why we would now allocate and capture this data into a field.  Why is that change being made. What data was found that indicated that such a change was appropriate or desirable?